### PR TITLE
Feature/add reference endpoint

### PR DESCRIPTION
**Pull request by hihilary**
1. add endpoints for trainee-reference
2. just leave the tests there for further config.

### CODE REVIEW

File: src/main/java/uk/nhs/hee/tis/trainee/reference/api/CollegeResource.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package uk.nhs.hee.tis.trainee.reference.api;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/api/CollegeResource.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/api/CollegeResource.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import uk.nhs.hee.tis.trainee.reference.dto.CollegeDto;
+import uk.nhs.hee.tis.trainee.reference.mapper.CollegeMapper;
+import uk.nhs.hee.tis.trainee.reference.model.College;
+import uk.nhs.hee.tis.trainee.reference.service.CollegeService;
+
+
+@RestController
+@RequestMapping("/api")
+public class CollegeResource {
+
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/api/CollegeResource.java]: Remove extra blank line.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/api/GenderResource.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package uk.nhs.hee.tis.trainee.reference.api;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/api/GenderResource.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/api/GenderResource.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import uk.nhs.hee.tis.trainee.reference.dto.GenderDto;
+import uk.nhs.hee.tis.trainee.reference.mapper.GenderMapper;
+import uk.nhs.hee.tis.trainee.reference.model.Gender;
+import uk.nhs.hee.tis.trainee.reference.service.GenderService;
+
+
+@RestController
+@RequestMapping("/api")
+public class GenderResource {
+
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/api/GenderResource.java]: Remove extra blank line.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/api/CollegeResource.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  private static final Logger log = LoggerFactory.getLogger(CollegeResource.class);
+  @Autowired
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/api/CollegeResource.java]: Prefer constructor injection over annotations.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/api/GenderResource.java [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  private static final Logger log = LoggerFactory.getLogger(GenderResource.class);
+  @Autowired
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/api/GenderResource.java]: Prefer constructor injection over annotations.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/api/LocalOfficeResource.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
+package uk.nhs.hee.tis.trainee.reference.api;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/api/LocalOfficeResource.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/api/LocalOfficeResource.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+import uk.nhs.hee.tis.trainee.reference.mapper.LocalOfficeMapper;
+import uk.nhs.hee.tis.trainee.reference.model.LocalOffice;
+import uk.nhs.hee.tis.trainee.reference.service.LocalOfficeService;
+
+@RestController
+@RequestMapping("/api")
+public class LocalOfficeResource {
+
+  private static final Logger log = LoggerFactory.getLogger(LocalOfficeResource.class);
+  @Autowired
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/api/LocalOfficeResource.java]: Prefer constructor injection over annotations.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/api/QualificationResource.java [hunk 9]
```diff
@@ -0,0 +1,36 @@
+package uk.nhs.hee.tis.trainee.reference.api;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/api/QualificationResource.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/api/QualificationResource.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+import uk.nhs.hee.tis.trainee.reference.mapper.QualificationMapper;
+import uk.nhs.hee.tis.trainee.reference.model.Qualification;
+import uk.nhs.hee.tis.trainee.reference.service.QualificationService;
+
+@RestController
+@RequestMapping("/api")
+public class QualificationResource {
+
+  private static final Logger log = LoggerFactory.getLogger(QualificationResource.class);
+  @Autowired
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/api/QualificationResource.java]: Prefer constructor injection over annotations.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/dto/CollegeDto.java [hunk 11]
```diff
@@ -0,0 +1,15 @@
+package uk.nhs.hee.tis.trainee.reference.dto;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/dto/CollegeDto.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/dto/GenderDto.java [hunk 12]
```diff
@@ -0,0 +1,15 @@
+package uk.nhs.hee.tis.trainee.reference.dto;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/dto/GenderDto.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/dto/LocalOfficeDto.java [hunk 13]
```diff
@@ -0,0 +1,16 @@
+package uk.nhs.hee.tis.trainee.reference.dto;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/dto/LocalOfficeDto.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/dto/QualificationDto.java [hunk 14]
```diff
@@ -0,0 +1,15 @@
+package uk.nhs.hee.tis.trainee.reference.dto;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/dto/QualificationDto.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/CollegeMapper.java [hunk 15]
```diff
@@ -0,0 +1,18 @@
+package uk.nhs.hee.tis.trainee.reference.mapper;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/CollegeMapper.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/GenderMapper.java [hunk 16]
```diff
@@ -0,0 +1,18 @@
+package uk.nhs.hee.tis.trainee.reference.mapper;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/GenderMapper.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/LocalOfficeMapper.java [hunk 17]
```diff
@@ -0,0 +1,18 @@
+package uk.nhs.hee.tis.trainee.reference.mapper;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/LocalOfficeMapper.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/QualificationMapper.java [hunk 18]
```diff
@@ -0,0 +1,18 @@
+package uk.nhs.hee.tis.trainee.reference.mapper;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/mapper/QualificationMapper.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/model/College.java [hunk 19]
```diff
@@ -0,0 +1,26 @@
+package uk.nhs.hee.tis.trainee.reference.model;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/model/College.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/model/Gender.java [hunk 20]
```diff
@@ -0,0 +1,25 @@
+package uk.nhs.hee.tis.trainee.reference.model;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/model/Gender.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/model/LocalOffice.java [hunk 21]
```diff
@@ -0,0 +1,26 @@
+package uk.nhs.hee.tis.trainee.reference.model;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/model/LocalOffice.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/model/Qualification.java [hunk 22]
```diff
@@ -0,0 +1,26 @@
+package uk.nhs.hee.tis.trainee.reference.model;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/model/Qualification.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/repository/CollegeRepository.java [hunk 23]
```diff
@@ -0,0 +1,9 @@
+package uk.nhs.hee.tis.trainee.reference.repository;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/repository/CollegeRepository.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/repository/LocalOfficeRepository.java [hunk 24]
```diff
@@ -0,0 +1,10 @@
+package uk.nhs.hee.tis.trainee.reference.repository;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/repository/LocalOfficeRepository.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/repository/GenderRepository.java [hunk 25]
```diff
@@ -0,0 +1,10 @@
+package uk.nhs.hee.tis.trainee.reference.repository;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/repository/GenderRepository.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/repository/QualificationRepository.java [hunk 26]
```diff
@@ -0,0 +1,9 @@
+package uk.nhs.hee.tis.trainee.reference.repository;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/repository/QualificationRepository.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/CollegeService.java [hunk 27]
```diff
@@ -0,0 +1,9 @@
+package uk.nhs.hee.tis.trainee.reference.service;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/CollegeService.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/GenderService.java [hunk 28]
```diff
@@ -0,0 +1,9 @@
+package uk.nhs.hee.tis.trainee.reference.service;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/GenderService.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/LocalOfficeService.java [hunk 29]
```diff
@@ -0,0 +1,9 @@
+package uk.nhs.hee.tis.trainee.reference.service;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/LocalOfficeService.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/QualificationService.java [hunk 30]
```diff
@@ -0,0 +1,9 @@
+package uk.nhs.hee.tis.trainee.reference.service;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/QualificationService.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/CollegeServiceImpl.java [hunk 31]
```diff
@@ -0,0 +1,19 @@
+package uk.nhs.hee.tis.trainee.reference.service.impl;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/CollegeServiceImpl.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/CollegeServiceImpl.java [hunk 32]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.nhs.hee.tis.trainee.reference.model.College;
+import uk.nhs.hee.tis.trainee.reference.repository.CollegeRepository;
+import uk.nhs.hee.tis.trainee.reference.service.CollegeService;
+
+@Service
+public class CollegeServiceImpl implements CollegeService {
+
+  @Autowired
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/CollegeServiceImpl.java]: Prefer constructor injection over annotations.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/GenderServiceImpl.java [hunk 33]
```diff
@@ -0,0 +1,19 @@
+package uk.nhs.hee.tis.trainee.reference.service.impl;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/GenderServiceImpl.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/GenderServiceImpl.java [hunk 34]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.nhs.hee.tis.trainee.reference.model.Gender;
+import uk.nhs.hee.tis.trainee.reference.repository.GenderRepository;
+import uk.nhs.hee.tis.trainee.reference.service.GenderService;
+
+@Service
+public class GenderServiceImpl implements GenderService {
+
+  @Autowired
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/GenderServiceImpl.java]: Prefer constructor injection over annotations.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/LocalOfficeServiceImpl.java [hunk 35]
```diff
@@ -0,0 +1,19 @@
+package uk.nhs.hee.tis.trainee.reference.service.impl;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/LocalOfficeServiceImpl.java]: No copyright notice.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/LocalOfficeServiceImpl.java [hunk 36]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.nhs.hee.tis.trainee.reference.model.LocalOffice;
+import uk.nhs.hee.tis.trainee.reference.repository.LocalOfficeRepository;
+import uk.nhs.hee.tis.trainee.reference.service.LocalOfficeService;
+
+@Service
+public class LocalOfficeServiceImpl implements LocalOfficeService {
+
+  @Autowired
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/LocalOfficeServiceImpl.java]: Prefer constructor injection over annotations.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/QualificationServiceImpl.java [hunk 37]
```diff
@@ -0,0 +1,19 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.nhs.hee.tis.trainee.reference.model.Qualification;
+import uk.nhs.hee.tis.trainee.reference.repository.QualificationRepository;
+import uk.nhs.hee.tis.trainee.reference.service.QualificationService;
+
+@Service
+public class QualificationServiceImpl implements QualificationService {
+
+  @Autowired
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/QualificationServiceImpl.java]: Prefer constructor injection over annotations.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/QualificationServiceImpl.java [hunk 38]
```diff
@@ -0,0 +1,19 @@
+package uk.nhs.hee.tis.trainee.reference.service.impl;
```
Comment (Judge40) [inline src/main/java/uk/nhs/hee/tis/trainee/reference/service/impl/QualificationServiceImpl.java]: No copyright notice.

File: src/test/java/uk/nhs/hee/tis/trainee/reference/TisTraineeReferenceApplicationTest.java [hunk 39]
```diff
@@ -27,8 +27,8 @@
 @SpringBootTest
 class TisTraineeReferenceApplicationTest {
 
-  @Test
-  void contextLoads() {
-
-  }
+//  @Test
```
Comment (Judge40) [inline src/test/java/uk/nhs/hee/tis/trainee/reference/TisTraineeReferenceApplicationTest.java]: Uncomment this, sonarcloud can be configured to ignore coverage of the application class but the test is still needed.

File: src/main/java/uk/nhs/hee/tis/trainee/reference/api/CollegeResource.java (hunk 5 above)
Comment (hihilary) [reply src/main/java/uk/nhs/hee/tis/trainee/reference/api/CollegeResource.java]: All the annotations has been modified to constructor injection.

**Pull request merged**